src/js/stores/MesosStateStore.js

```diff
@@ -256,7 +256,13 @@ class MesosStateStore extends GetSetBaseStore {
     }
 
     return tasks
-      .filter(task => task.isStartedByMarathon && task.name === mesosTaskName)
+      .filter(
+        task =>
+          task.isStartedByMarathon &&
+          task.name === mesosTaskName &&
+          task.discovery &&
```

Comment:
According to protobufs `discovery` is optional https://github.com/apache/mesos/blob/40210dbcfb218d6ef3f647b047afbdda24133f4d/include/mesos/mesos.proto#L2184

Reply:
So what is the best way to distinguish service task from a framework task?

Reply:
That's the core of the issue I'm afraid. I don't think I have a definitive answer.